### PR TITLE
[#3430] improvement(hdfs):  Add a docker kerberized CI image to test kerberos authentication 

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Add a docker kerberosized ci image.

### Why are the changes needed?

We need this image to test kerberos authentication for HDFS.

Fix: #3430

### Does this PR introduce _any_ user-facing change?

N/A.

### How was this patch tested?

Test locally, IT test in the next PR will confirm it.

### CODE REVIEW

File: docs/docker-image-details.md [hunk 1]
```diff
@@ -196,3 +196,13 @@ Changelog
   - Expose ports:
     - `8030` Doris FE HTTP port
     - `9030` Doris FE MySQL server port
+
+## Gravitino CI kerberosized HDFS image
+
+You can use this image to test kerberized HDFS.
+
+Changelog
+- gravitino-ci-kerberosized-hdfs:0.1.0
```
Comment (qqqttt123) [inline docs/docker-image-details.md:205]: kerberos-hdfs.

File: dev/docker/build-docker.sh [hunk 2]
```diff
@@ -74,6 +74,9 @@ fi
 if [[ "${component_type}" == "hive" ]]; then
   . ${script_dir}/hive/hive-dependency.sh
   build_args="--build-arg HADOOP_PACKAGE_NAME=${HADOOP_PACKAGE_NAME} --build-arg HIVE_PACKAGE_NAME=${HIVE_PACKAGE_NAME} --build-arg JDBC_DIVER_PACKAGE_NAME=${JDBC_DIVER_PACKAGE_NAME}"
+elif [[ "${component_type}" == "hdfs" ]]; then
```
Comment (qqqttt123) [inline dev/docker/build-docker.sh:77]: kerberos-hdfs.

File: dev/docker/hdfs/yarn-env.sh [hunk 3]
```diff
@@ -0,0 +1,124 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
```
Comment (qqqttt123) [inline dev/docker/hdfs/yarn-env.sh:1]: This is useless for yarn.

Comment (yuqi1129) [reply dev/docker/hdfs/yarn-env.sh:1]: So we can remove it freely?

**Comment by jerryshao** [comment]
Shall we have a kerberized hdfs or we add HMS also?

**Comment by yuqi1129** [comment]
> Shall we have a kerberized hdfs or we add HMS also?

As the version of HDFS in HMS and HDFS clusters is not the same( v2.6 & v3.1), I'm afraid that we cannot merge them and both are necessary. For the kerberized HMS, @qqqttt123 has prepared a draft PR for it.

**Comment by yuqi1129** [comment]
> Why do we choose Hadoop 3.1 rather than Hadoop 2.6?

The version of Hadoop in `Hadoop-catalog` is 3.1, so I created a Hadoop 3 cluster to test its functionality. I'm not completely sure Hadoop 2.6 is suitable for this module.

**Comment by yuqi1129** [comment]
@qqqttt123 
Would you like to take another look at the document?

**Comment by jerryshao** [comment]
Another thing is that you still need to have a kerberized image for Hive, so this will be a duplicated work because you need to do it again when you test hive catalog with Keberos enabled.

**Comment by yuqi1129** [comment]
> Another thing is that you still need to have a kerberized image for Hive, so this will be a duplicated work because you need to do it again when you test hive catalog with Keberos enabled.

The question is: whether we need to provide a different version of HDFS such as Hadoop 3 for the Hadoop catalog, if this is unnecessary, I think @qqqttt123 should move forward https://github.com/datastrato/gravitino/pull/2364 and I can reused it.

**Comment by jerryshao** [comment]
> whether we need to provide a different version of HDFS such as Hadoop 3 for the Hadoop catalog
It's not necessary to have a separate image for Hadoop catalog. I've already used Hive image for IT, we can reuse it for kerberos enabled test.

> I think @qqqttt123 should move forward https://github.com/datastrato/gravitino/pull/2364 and I can reused it.
I don't think @qqqttt123 has enough time to do so. It would be good if you can continue on his work.

**Comment by jerryshao** [comment]
This is not needed, right? @yuqi1129 can you please close this PR?

**Comment by yuqi1129** [comment]
> This is not needed, right? @yuqi1129 can you please close this PR?

Yeah, I will close it.

**Comment by yuqi1129** [comment]
It's not necessary as #2368 has been merged.

**Pull request closed without merging**